### DISCUSSION TITLE
Doodba and Odoo.sh

### DISCUSSION THREAD

**Discussion by ubidev** [Tecnativa/doodba#612] [General]
Hi everyone!
After going through the different sections of this very interesting project and sucessfully deploying it under wsl/docker on Windows, I'm still wondering though how can it be integrated with odoo.sh. Can it be done and is it worth the trouble? What is your take on this.

Thanks in advance for taking the time and have a great week :)
UBiDEV

**Comment by pedrobaeza** [comment]
I'm afraid you can't run Doodba on Odoo.sh, as it's a restricted hosting that has its own rules and doesn't accept Docker nor the needed stuff for run it.

**Comment by ubidev** [comment]
I understand,
But could I use Doodba to dev/test in a local instance then push the changes into a github branch that is used by odoo.sh then from there push it to staging then prod?
My goal here is to avoid dev/testing directly in odoo.sh for now as my customer's workers load is saturated and he does not want to by more workers nor move to a dedicated server/vm.
Thanks again :)

**Reply by pedrobaeza** [reply]
You can always use Doodba in local for development, but AFAIK you should redo part of the work at Odoo.sh, as if I'm not wrong, the configuration of the repositories to use is done through UI, and if possible by files, you should need a tool for converting Doodba configuration files (essentially repos.yaml and addons.yaml) to the Odoo.sh format. Anyway, I don't know too much Odoo.sh to say its features, and more lately.

**Reply by ubidev** [reply]
Thanks for the insight !
Since Doodba is soo fantastic to provision local dev environment, I'm going to try it with Odoo.sh using some workflow like this as a starter:

1. git pull odoo.sh dev branch in sub folder of copier-template
2. set proper gitignore to skip .git folder whithin dev folder
3. dev/test module in doodba copier-template
5. git commit,push dev folder to Odoo.sh
6. push Odoo.sh dev branch changes to staging branch the to prod if alll good!

If that works,  I good have Doodba for local, cloud and odoo.sh development! :)

What do you think? Feasable?

Have an amazing day :)

**Reply by pedrobaeza** [reply]
Yeah, if that works, you can provide some documentation to be included in the project for those of you using Odoo.sh.

**Reply by ubidev** [reply]
You can count on me, I definately will :-)

Talking about documentation :) How do I setup doodba to install the full accounting module? I tried/failed miserably via addons.yaml and even with "addons -w account_accountant -w account -W pad -W pad_project -W note_pad -i -e -n -c init"

every other modules install and are accessible in the odoo gui but not accounting :(

What did I miss

**Comment by JordiBForgeFlow** [comment]
@ubidev you can use https://pypi.org/project/odoo-filter-addons/ to create a repository where your modules are aggregated based on a repos.yaml and addons.yaml outside of doodba, and then use the resulting repo as the shared one for odoo.sh and doodba (where in doodba you refer to this repo in the doodba's repos.yaml).

**Reply by ubidev** [reply]
Thanks for this great suggestion.
 I'll create a new discussion so I can find out/understand why technically, the full accounting module is not installed when referencing it from addons.yaml or via the addons script

Thanks guys :)

**Comment by JordiBForgeFlow** [comment]
@ubidev when a module is in the addons.yaml it will be available for install. But you have to install it yourself.
